lib/grinder_files.dart

```diff
@@ -122,19 +322,18 @@ String fileExt(FileSystemEntity entity) {
  */
 String baseName(FileSystemEntity entity) {
   String name = entity.path;
-  int index = name.lastIndexOf(Platform.pathSeparator);
-
+  int index = name.lastIndexOf(_sep);
   return (index != -1 ? name.substring(0, index) : null);
 }
 
 File joinFile(Directory dir, List<String> files) {
```

Comment:
If we add `Path`, maybe this is easy enough: `new Path(directory).join(...).asDirectory`?